### DISCUSSION TITLE
Battery friendly player

### DISCUSSION THREAD

**Discussion by divyansh10101** [moneytoo/Player#50] [General]
Which player consumes least battery.

**Comment by moneytoo** [comment]
This player app uses `ExoPlayer` for video playback so the battery consumption should be very little. The most battery friendly video playback engine should be `MediaPlayer` - the one that's built in the Android itself and is the one most optimized (hw acceleration) - but also limited in some ways (like variety of supported audio decoders). ExoPlayer adds lots of features while [keeping similar battery consumption](https://exoplayer.dev/battery-consumption.html) as it uses the same hw video decoders.

As mentioned in the article from ExoPlayer developers, Just Player keeps using `SurfaceView` which is more battery friendly (uses hw overlay).  `SurfaceView` is also used when zooming the video. This means that this feature is limited to newer Android devices (7+) but it allows lower battery consumption.
